packages/server/src/adapters/aws-lambda/index.ts

```diff
@@ -50,11 +50,26 @@ function getHTTPMethod(event: APIGatewayEvent) {
     message: UNKNOWN_PAYLOAD_FORMAT_VERSION_ERROR_MESSAGE,
   });
 }
+
 function getPath(event: APIGatewayEvent) {
   if (isPayloadV1(event)) {
+    const matches = event.resource.matchAll(/\{(.*?)\}/g);
+    for (const match of matches) {
+      const group = match[1];
+      if (group.includes('+') && event.pathParameters) {
+        return event.pathParameters[group.replace('+', '')] || '';
+      }
+    }
     return event.path.slice(1);
   }
   if (isPayloadV2(event)) {
+    const matches = event.routeKey.matchAll(/\{(.*?)\}/g);
```

Comment:
Wouldn’t it make sense to extract the shared code here into a function taking routeKey or resource depending on the payload version?

Reply:
All the options to refactor here feel like they makes the code more complicated not less which is why its just duplicated right now, if you have any suggestions happy to take them on board :)